### DISCUSSION TITLE
Making Github Pages work with <script type="module">

### DISCUSSION THREAD

**Discussion by LucasRegalar** [community/community#112227] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello there,

is there a way to host the GitHub Page while using type="module" in my <script> element?

I have the type="module" so I can split my JS File in multiple files. 

Greetings

Lucas

**Comment by kuciapakrystian** [comment, marked as answer]
Yes, you can host a GitHub Page while using the type="module" attribute in your <script> element. However, there are a few things to keep in mind:

Server Compatibility: GitHub Pages support serving static content, including HTML, CSS, and JavaScript files. As of my last update, GitHub Pages fully supports serving JavaScript modules (type="module").

File Structure: Ensure that your JavaScript files are structured properly to take advantage of module imports and exports. Each JavaScript file should contain a module, and you can import these modules into your main JavaScript file.

HTML Markup: In your HTML file, you'll need to include your main JavaScript file using a <script> tag with the type="module" attribute. For example:

html
Copy code
<script type="module" src="main.js"></script>
Cross-Origin Policy: If you're fetching resources from other origins, be mindful of CORS (Cross-Origin Resource Sharing) policies. GitHub Pages generally serves content from the same origin, so you might encounter issues if you're trying to fetch resources from different origins.

Testing Locally: Before deploying to GitHub Pages, it's a good idea to test your setup locally. You can use a local development server like http-server or live-server to serve your files locally and test whether the modules are loading correctly.

Here's a basic example of how your file structure might look:
project-root/
│
├── index.html
├── main.js
├── module1.js
└── module2.js
And your index.html might look like:
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>My GitHub Page</title>
</head>
<body>
    

    
    <script type="module" src="main.js"></script>
</body>
</html>

And in your main.js, you might have imports like:

import { someFunction } from './module1.js';
import { anotherFunction } from './module2.js';

// Use the imported functions
someFunction();
anotherFunction();

Ensure that your file paths in the src attributes of your script tags are correct relative to the HTML file they're included in. With this setup, you should be able to host your GitHub Page with JavaScript modules successfully.

**Reply by LucasRegalar** [reply]
Not quite sure what was the problem, but it is working now! 
Thank you for the help :)
